sklearn/utils/_array_api.py

```diff
@@ -123,43 +137,36 @@ def get_namespace(*arrays):
     Returns
     -------
     namespace : module
-        Namespace shared by array objects.
+        Namespace shared by array objects. If any of the `arrays` are not arrays,
+        the namespace defaults to NumPy.
 
     is_array_api : bool
         True of the arrays are containers that implement the Array API spec.
     """
-    # `arrays` contains one or more arrays, or possibly Python scalars (accepting
-    # those is a matter of taste, but doesn't seem unreasonable).
-    # Returns a tuple: (array_namespace, is_array_api)
-
-    if not get_config()["array_api_dispatch"]:
-        return _NumPyApiWrapper(), False
+    return _get_namespace(
+        *arrays, array_api_dispatch=get_config()["array_api_dispatch"]
+    )
 
-    namespaces = {
-        x.__array_namespace__() if hasattr(x, "__array_namespace__") else None
-        for x in arrays
-        if not isinstance(x, (bool, int, float, complex))
-    }
 
-    if not namespaces:
-        # one could special-case np.ndarray above or use np.asarray here if
-        # older numpy versions need to be supported.
-        raise ValueError("Unrecognized array input")
+def _get_namespace(*arrays, array_api_dispatch=False):
+    if not array_api_dispatch:
+        return array_api_compat_numpy, False
+    try:
+        namespace, is_array = array_api_compat.get_namespace(*arrays), True
+    except TypeError as e:
+        if str(e).startswith("The input is not a supported array type"):
+            return array_api_compat_numpy, False
+        raise
 
-    if len(namespaces) != 1:
-        raise ValueError(f"Multiple namespaces for array inputs: {namespaces}")
+    if namespace.__name__ in {"numpy.array_api", "cupy.array_api"}:
+        namespace = _ArrayAPIWrapper(namespace)
 
-    (xp,) = namespaces
-    if xp is None:
-        # Use numpy as default
-        return _NumPyApiWrapper(), False
-
-    return _ArrayAPIWrapper(xp), True
+    return namespace, is_array
```

Comment:
I think that `is_array` should be renamed to something more specific, such as `is_array_api_compliant`.

The local variables used to store this return values in callers should be updated accordingly.